server.go

```diff
@@ -295,26 +303,76 @@ type Server struct {
 	DecorateReader DecorateReader
 	// DecorateWriter is optional, allows customization of the process that writes raw DNS messages.
 	DecorateWriter DecorateWriter
-
+	// UDP packet or TCP connection queue
+	queue chan *response
+	// Workers count
+	workersCount int32
 	// Shutdown handling
 	lock    sync.RWMutex
 	started bool
 }
 
+func (srv *Server) worker(w *response) {
+	workersCount := atomic.LoadInt32(&srv.workersCount)
+	if workersCount > maxWorkersCount {
+		w.Close()
+		return
+	}
+	atomic.AddInt32(&srv.workersCount, 1)
+
+	srv.serve(w)
+
+	count := 0
+	timeout := time.After(idleWorkerTimeout)
+LOOP:
+	for {
+		select {
+		case w, ok := <-srv.queue:
+			if !ok {
+				break LOOP
+			}
+			count += 1
+			srv.serve(w)
+		case <-timeout:
+			if count == 0 {
+				break LOOP
+			}
+			count = 0
+			timeout = time.After(idleWorkerTimeout)
```

Comment:
It should be safe to call `timeout.Reset(...)` here paired with `timer.NewTicker` (? - from memory) instead of creating a new timer each iteration.

Reply:
Done with Ticker - don't follow about Reset()

Reply:
I was thinking of `time.NewTimer` + `Reset` - I didn’t check the docs.

I don’t think `time.NewTicker` is right here, because it drops channel sends on the floor, meaning this would be sutbley wrong.